src/mongocrypt-kms-ctx.c

```diff
@@ -110,7 +111,13 @@ _init_common (mongocrypt_kms_ctx_t *kms,
               _mongocrypt_log_t *log,
               _kms_request_type_t kms_type)
 {
-   kms->parser = kms_response_parser_new ();
+   if (kms_type == MONGOCRYPT_KMS_KMIP_REGISTER ||
```

Comment:
Seeing that this gets repeated a couple times, it might be worth considering making it an is_kms() predicate. OTOH, it it's also on the edge of just being noise to do so-- up to you.